lib/forspell/reporter.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'pastel'
+require 'logger'
+
+module Forspell
+  class Reporter
+    SUCCESS_CODE = 0
+    ERROR_CODE = 1
+
+    attr_reader :total_errors
+
+    def initialize(logfile:,
+                   verbose:,
+                   format:,
+                   group:)
+
+      FileUtils.touch(logfile) if logfile.is_a?(String)
+      @logger = Logger.new(logfile || STDOUT)
+      @error_logger = Logger.new(STDERR)
+
+      [@logger, @error_logger].each do |logger|
+        logger.formatter = proc do |_severity, _datetime, _progname, msg|
+          "#{msg}\n"
+        end
+      end
+
+      @verbose = verbose
+      @format = group ? 'group' : format
+
+      @pastel = Pastel.new(enabled: $stdout.tty?)
+      @errors = []
+    end
+
+    def file(path)
+      @logger.info "Processing #{path}" if @verbose
```

Comment:
Logging levels are exactly for this (output or suppress different messages), so you don't need neither `@verbose` nor keeping _two_ logs.
It is just `@logger.info "processing...."` and `@logger.error "while reading that file..."`
And at the beginning, you set `@logger.level` to appropriate setting (if it is verbose, you set it to `INFO` -- it will output "Processing..." messages, and otherwise you set it, for ex., to `WARNING`, and then it will swallow them silently)